### ISSUE TITLE
MiniCPM-LLaMA3-V2.5 fine-tuning error

### ISSUE THREAD

**Issue by yipoh** [OpenBMB/MiniCPM-V#128]
Very meaningful work. On our [AesBench,](https://github.com/yipoh/AesBench) we tested MiniCPM-LLaMA3-V2.5, which became the only open source model to surpass GPT-4o. However, we fine-tuned it according to the reference provided and encountered an error, which appeared to be caused by inconsistent dimensions.  How can I solve this problem?  Thanks a lot.

![e5286715c1d4734463d8435fa57eaa2](https://github.com/OpenBMB/MiniCPM-V/assets/124764227/d8d3ec76-6f22-4538-be38-aef509492dea)

**Comment by qyc-98** [comment]
Hello, in order to more accurately locate and resolve the issue you are encountering, we need the data input that was present when the error occurred. This will help us reproduce the problem and conduct debugging.

**Comment by dylanqyuan** [comment]
> Hello, in order to more accurately locate and resolve the issue you are encountering, we need the data input that was present when the error occurred. This will help us reproduce the problem and conduct debugging.

Hey Qianyu, 感谢回复 :smile:
以下是实验的相关设置：
GPUs: A40*8, 

`cd MiniCPM-V`
`mv finetune/* .`
`sh finetune_ds.sh`

其中`finetune_ds.sh`内容如下：
`#!/bin/bash

> GPUS_PER_NODE=8
NNODES=1
NODE_RANK=0
MASTER_ADDR=localhost
MASTER_PORT=6001

MODEL="openbmb/MiniCPM-Llama3-V-2_5" # or openbmb/MiniCPM-V-2
DATA="XXX"
EVAL_DATA="XXX"
LLM_TYPE="llama3" # if use openbmb/MiniCPM-V-2, please set LLM_TYPE=minicpm

DISTRIBUTED_ARGS="
    --nproc_per_node $GPUS_PER_NODE \
    --nnodes $NNODES \
    --node_rank $NODE_RANK \
    --master_addr $MASTER_ADDR \
    --master_port $MASTER_PORT
"
torchrun $DISTRIBUTED_ARGS finetune.py \
    --model_name_or_path $MODEL \
    --llm_type $LLM_TYPE \
    --data_path $DATA \
    --eval_data_path $EVAL_DATA \
    --remove_unused_columns false \
    --label_names "labels" \
    --prediction_loss_only  false \
    --bf16 true \
    --bf16_full_eval true \
    --do_train \
    --do_eval \
    --model_max_length 2048 \
    --max_steps 80000 \
    --eval_steps 200 \
    --output_dir output/output_minicpmv2 \
    --logging_dir output/output_minicpmv2 \
    --logging_strategy "steps" \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "steps" \
    --save_strategy "steps" \
    --save_steps 1000 \
    --save_total_limit 10 \
    --learning_rate 5e-7 \
    --weight_decay 0.1 \
    --adam_beta2 0.95 \
    --warmup_ratio 0.01 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --gradient_checkpointing True \
    --deepspeed ds_config_zero2.json \
    --report_to "tensorboard" # wandb


    
    其中的指令训练集XXX.json的格式如下：
    `[
    {
      "id": "0",
      "image": 'path/to/image_0.jpg',
      "conversations": [
            {
              'role': 'user', 
              'content': '<image>\nHow many desserts are on the white plate?'
            }, 
            {
                'role': 'assistant', 
                'content': 'There are three desserts on the white plate.'
            }  
        ]
    },
  ]` 
  
  错误提示如下图：
![微信图片_20240523142757](https://github.com/OpenBMB/MiniCPM-V/assets/63891238/e445409a-bc53-48ed-835c-031965aaa19d)


另外还有一个PyTorch版本不一致的问题，
requirements.txt(https://github.com/OpenBMB/MiniCPM-V/blob/main/requirements.txt)中PyTorch版本是2.1.2，运行`finetune_ds.sh`时候有如下的warning:
`sparse_attn requires a torch version >= 1.4and <2.0 but detect 2.1.2`

best,

**Comment by qyc-98** [comment]
hi my email is socqyc@gmail.com. thanks！
